tests/provider: Unknown operation CreateReportGroup (GovCloud)

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

These tests fails on GovCloud:

```

TestAccAWSCodeBuildReportGroup_basic
TestAccAWSCodeBuildReportGroup_disappears
TestAccAWSCodeBuildReportGroup_export_s3
TestAccAWSCodeBuildReportGroup_tags
```

With this error: `InvalidInputException: Unknown operation CreateReportGroup`.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_codebuild_report_group

### Terraform Configuration Files

Example config:

```hcl
resource "aws_codebuild_report_group" "test" {
  name = "yak_report_group"
  type = "TEST"

  export_config {
    type = "S3"

    s3_destination {
      bucket              = aws_s3_bucket.test.id
      encryption_key      = aws_kms_key.test.arn
      encryption_disabled = false
      packaging           = "ZIP"
      path                = "/some2"
    }
  }
}
```

### Debug Output

```
TestAccAWSCodeBuildReportGroup_basic: resource_aws_codebuild_report_group_test.go:20: Step 1/2 error: Error running apply: 2020/10/23 06:41:37 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: error creating CodeBuild Report Groups: InvalidInputException: Unknown operation CreateReportGroup

--- FAIL: TestAccAWSCodeBuildReportGroup_basic (11.09s)
```

### References

* #0000

--- a/aws/provider_test.go
+++ b/aws/provider_test.go
@@ -868,6 +868,9 @@ func testAccPreCheckSkipError(err error) bool {
 	if isAWSErr(err, "UnsupportedOperation", "") {
 		return true
 	}
+	if isAWSErr(err, "InvalidInputException", "Unknown operation") {
+		return true
+	}
 	return false
 }
 

--- a/aws/resource_aws_codebuild_report_group_test.go
+++ b/aws/resource_aws_codebuild_report_group_test.go
@@ -18,7 +18,7 @@ func TestAccAWSCodeBuildReportGroup_basic(t *testing.T) {
 	resourceName := "aws_codebuild_report_group.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSCodeBuild(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSCodeBuildReportGroup(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSCodeBuildReportGroupDestroy,
 		Steps: []resource.TestStep{
@@ -48,7 +48,7 @@ func TestAccAWSCodeBuildReportGroup_export_s3(t *testing.T) {
 	resourceName := "aws_codebuild_report_group.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSCodeBuild(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSCodeBuildReportGroup(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSCodeBuildReportGroupDestroy,
 		Steps: []resource.TestStep{
@@ -96,7 +96,7 @@ func TestAccAWSCodeBuildReportGroup_tags(t *testing.T) {
 	resourceName := "aws_codebuild_report_group.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSCodeBuild(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSCodeBuildReportGroup(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSCodeBuildReportGroupDestroy,
 		Steps: []resource.TestStep{
@@ -140,7 +140,7 @@ func TestAccAWSCodeBuildReportGroup_disappears(t *testing.T) {
 	resourceName := "aws_codebuild_report_group.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSCodeBuild(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSCodeBuildReportGroup(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSCodeBuildReportGroupDestroy,
 		Steps: []resource.TestStep{
@@ -154,6 +154,22 @@ func TestAccAWSCodeBuildReportGroup_disappears(t *testing.T) {
 			},
 		},
 	})
+}
+
+func testAccPreCheckAWSCodeBuildReportGroup(t *testing.T) {
+	conn := testAccProvider.Meta().(*AWSClient).codebuildconn
+
+	input := &codebuild.ListReportGroupsInput{}
+
+	_, err := conn.ListReportGroups(input)
+
+	if testAccPreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
 }
 
 func testAccCheckAWSCodeBuildReportGroupDestroy(s *terraform.State) error {

